engine/engine.go

```diff
@@ -342,6 +342,14 @@ func resolve(dbp zesty.DBProvider, res *resolution.Resolution, t *task.Task, deb
 		case s := <-stepChan:
 			s.LastRun = time.Now()
 
+			// Replace task's tags with the tags returned in the step.
```

Comment:
I think the plugin should return a single map of tags, with the zero value interpreted as a delete instruction.
It may seem counter-intuitive but it's a pretty simple protocol and it lets template writers make add-or-delete decisions with templating instructions.

Reply:
@loopfz We've discussed about this with @rbeuque74 and I disagree for a simple reason: a bug/error in the templating system that would result in a variable/input to "render" as an empty string would cause a tag to be deleted by accident. Not only I find it counter-intuitive, but also potentially risky in such case.

Reply:
There are way more severe templating issues, e.g. doing the wrong request in the first place.
This is a non issue imo, and the current implementation makes it very static, which isnt a desirable trait.